protocols/directdefund/directdefund.go

```diff
@@ -214,6 +214,10 @@ func (o *Objective) UpdateWithChainEvent(event chainservice.Event) (protocols.Ob
 				updated.C.OnChainFunding = e.Holdings.Clone()
 			}
 		}
+	case chainservice.DepositedEvent:
+		// TODO: This is a workaround for https://github.com/statechannels/go-nitro/issues/735
+		// Eventually we should handle this by checking block numbers
```

Comment:
Can you clarify how that solution will work? Is there anything "wrong" this approach / could we remove this todo?

Reply:
Good point, I think this is a probably a valid way to deal with it.